src/main/java/igrad/model/CourseBook.java

```diff
@@ -0,0 +1,120 @@
+package igrad.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import igrad.model.module.Module;
+import igrad.model.module.UniqueModuleList;
+import javafx.collections.ObservableList;
+
+/**
+ * Wraps all data at the course-book level
+ * Duplicates are not allowed (by .isSameModule comparison)
+ */
+public class CourseBook implements ReadOnlyCourseBook {
+
+    private final UniqueModuleList modules;
+
+    /*
+     * The 'unusual' code block below is a non-static initialization block, sometimes used to avoid duplication
+     * between constructors. See https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html
+     *
+     * Note that non-static init blocks are not recommended to use. There are other ways to avoid duplication
+     *   among constructors.
+     */
+    {
+        modules = new UniqueModuleList();
+    }
+
+    public CourseBook() {}
+
+    /**
+     * Creates an CourseBook using the Persons in the {@code toBeCopied}
+     */
+    public CourseBook(ReadOnlyCourseBook toBeCopied) {
+        this();
+        resetData(toBeCopied);
+    }
+
+    //// list overwrite operations
```

Comment:
i'm not sure about these comments tho, `////`, i only know these `//`, but its probably in the original code-base as well, would fix that too.